DepotDownloader/ContentDownloader.cs

```diff
@@ -1295,5 +1281,49 @@ private static async Task DownloadSteam3Async(uint appId, List<DepotDownloadInfo
             }
 
         }
+
+        static void DumpManifestToTextFile( DepotDownloadInfo depot, ProtoManifest manifest )
+        {
+            StringBuilder manifestBuilder = new StringBuilder();
+            string txtManifest = Path.Combine( depot.installDir, string.Format( "manifest_{0}_{1}.txt", depot.id, depot.manifestId ) );
+            manifestBuilder.Append( string.Format( "Content Manifest for Depot {0}\n\n", depot.id ) );
+            manifestBuilder.Append( string.Format( "Manifest ID / date     : {0} / {1}\n", depot.manifestId, manifest.CreationTime ) );
+            
+            int numFiles = 0, numChunks = 0;
+            ulong uncompressedSize = 0, compressedSize = 0;
+
+            foreach ( var file in manifest.Files )
+            {
+                if ( file.Flags.HasFlag( EDepotFileFlag.Directory ) )
+                    continue;
+
+                numFiles++;
+                numChunks += file.Chunks.Count;
+
+                foreach ( var chunk in file.Chunks )
+                {
+                    uncompressedSize += chunk.UncompressedLength;
+                    compressedSize += chunk.CompressedLength;
+                }
+            }
+
+            manifestBuilder.Append( string.Format( "Total number of files  : {0}\n", numFiles ) );
+            manifestBuilder.Append( string.Format( "Total number of chunks : {0}\n", numChunks ) );
+            manifestBuilder.Append( string.Format( "Total bytes on disk    : {0}\n", uncompressedSize ) );
+            manifestBuilder.Append( string.Format( "Total bytes compressed : {0}\n", compressedSize ) );
+            manifestBuilder.Append( "\n          Size Chunks File SHA                                 Flags Name\n" );
+
+            foreach ( var file in manifest.Files )
+            {
+                manifestBuilder.Append( string.Format( "{0} {1} {2} {3} {4}\n",
+                    file.TotalSize.ToString().PadLeft( 14, ' ' ),
+                    file.Chunks.Count.ToString().PadLeft( 6, ' ' ),
+                    BitConverter.ToString( file.FileHash ).Replace( "-", "" ),
+                    ((int)(file.Flags)).ToString( "X" ).PadLeft( 5, ' ' ),
+                    file.FileName.ToString() ) );

```

Comment:
Don't call `ToString()` on an object that is already a string.

Reply:
Again, what to do with SHA-1 hash in this formatting method?

Reply:
extract it to a variable?